app/scripts/directives/payment.js

```diff
@@ -58,26 +58,94 @@ angular.module('confRegistrationWebApp')
-                  paymentErrors.push('Please enter the name on your card.');
-                }
-                var validationError = ccp.validateCardNumber(currentPayment.creditCard.number || '');
-                if(validationError){
-                  paymentErrors.push('Please enter a valid card number. The ' + validationError + '.');
-                }
-                if(!currentPayment.creditCard.expirationMonth || !currentPayment.creditCard.expirationYear){
-                  paymentErrors.push('Please enter your card expiration date.');
-                }
-                validationError = ccp.validateCardSecurityCode(currentPayment.creditCard.cvvNumber || '');
-                if(validationError){
-                  paymentErrors.push('Please enter a valid card security code. The ' + validationError + '.');
-                }
+                var validations = {
+                  nameOnCard: [
+                    {
+                      validate: required,
+                      errorMessage: 'You must enter the name on the card.'
+                    }
+                  ],
+                  number: [
+                    {
+                      validate: required,
+                      errorMessage: gettextCatalog.getString('You must enter a card number.')
+                    }, {
+                      validate: cruPayments.card.validate.minLength,
+                      errorMessage: gettextCatalog.getString('This card number must contain at least 13 digits.')
+                    }, {
+                      validate: cruPayments.card.validate.maxLength,
+                      errorMessage: gettextCatalog.getString('This card number cannot contain more than 16 digits.')
+                    }, {
+                      validate: cruPayments.card.validate.knownType,
+                      errorMessage: gettextCatalog.getString('This card type is not recognized.')
+                    }, {
+                      validate: cruPayments.card.validate.typeLength,
+                      get errorMessage() {
+                        var cardNumber = currentPayment.creditCard.number;
+                        var cardType = cruPayments.card.info.type(cardNumber);
+                        var expectedLength = cruPayments.card.info.expectedLengthForType(cardNumber) || [];
+                        var expectedDigits = expectedLength.join(gettextCatalog.getString(' or '));
+                        return gettextCatalog.getString('This is an invalid {{cardType}} number. It should have {{expectedDigits}} digits.', { cardType: cardType, expectedDigits: expectedDigits });
+                      }
+                    }, {
+                      validate: cruPayments.card.validate.checksum,
+                      errorMessage: gettextCatalog.getString('This card number is invalid. At least one digit was entered incorrectly.')
+                    }
+                  ],
+                  cvvNumber: [
```

Comment:
In cru-payments `1.1.1` I added one more of these. https://www.npmjs.com/package/cru-payments#crupaymentscreditcardcvvvalidatecardtypelengthcvv-cardtype

American Express apparently requires the CVV to be 4 digits